### DISCUSSION TITLE
Use HashMap<String, String> for aliases?

### DISCUSSION THREAD

**Discussion by brandonryan** [dansun-games/universe#3] [Ideas]
We don't really need to store a struct for these because the only information is what the alias is for.

Should make a new rust type alias so we can document what we are doing a bit better.
Something like:
```rs
/// Key: alias name.
/// Value: type name.
pub type AliasMap = HashMap<String, String>;
```

You could probably make a similar argument for all the rest of the types storing their names in the struct.
Needs further discussion.
Could do something like:
```rs
pub type NameMap<T> = HashMap<String, T>;
```

Really we are just talking about some memory footprint improvements. Maybe overthinking things.
@JoshuahBoardman thoughts?

**Comment by JoshuahBoardman** [comment]
Ya I agree, there isn't much value in duplicating that information. It only slightly changes the way we interact with the data:
```rust
for (name, desc) in self.structs.value() {
    write_struct(&mut file, name, desc)?;
}
```
I can make the changes if you want @brandonryan ?

**Comment by brandonryan** [comment]
Give me like 2 hours to finish my task up because we are going to conflict pretty bad I think
